### PR TITLE
{Role} Bump ResourceType.MGMT_AUTHORIZATION to 2020-04-01-preview

### PR DESCRIPTION
**Pull request by jiasli**
**Description**  

1. Bump `ResourceType.MGMT_AUTHORIZATION` to `2020-04-01-preview`.
2. Fix live tests for **Role** module.

### CODE REVIEW

**Comment by jiasli** [comment]
`test_create_for_rbac_with_existing_kv_cert` fails, because:

Azure CLI pinned `azure-devtools` to 1.0.0, which has a bug with `SingleValueReplacer` (fixed in https://github.com/Azure/azure-python-devtools/pull/56, but not released even until 1.1.1) at this line

https://github.com/Azure/azure-python-devtools/blob/1.0.0/src/azure_devtools/scenario_tests/preparers.py#L108

```py
body = str(request.body)
```

This is the wrong way to convert `bytes` to `str`:

https://docs.python.org/3/library/stdtypes.html#str

> Passing a `bytes` object to `str()` without the encoding or errors arguments falls under the first case of returning the informal string representation (see also the -b command-line option to Python)> . For example:
> 
> ```py
> >>> str(b'Zoot!')
> "b'Zoot!'"
> ```

This cause all the weird `request.body` in recorded YAMLs:

https://github.com/Azure/azure-cli/blob/55d53ba5f67bbfde5edba27d4689abd5aba709d6/src/azure-cli/azure/cli/command_modules/resource/tests/latest/recordings/test_generic_subscription_locks.yaml#L303-L304

Due to another bug in `vcrpy` (https://github.com/kevin1024/vcrpy/issues/543), replacers are called twice. When `<Request (POST) https://graph.windows.net/00000000-0000-0000-0000-000000000000/applications?api-version=1.6>` is recorded, the replacers are executed in the following order:

```
azure.cli.testsdk.utilities.GraphClientPasswordReplacer
azure.cli.testsdk.preparers.ResourceGroupPreparer        << This corrupts the JSON

azure.cli.testsdk.utilities.GraphClientPasswordReplacer  << This fails with JSONDecodeError
azure.cli.testsdk.preparers.ResourceGroupPreparer
```

As `ResourceGroupPreparer` inherits from `SingleValueReplacer`, it corrupts the JSON as

```
b'{"availableToOtherTenants": false, ...}'
```

Then `GraphClientPasswordReplacer` fails to load the JSON with `JSONDecodeError`:

https://github.com/Azure/azure-cli/blob/a30596720aae9995a125e84f3bb462e22dc55868/src/azure-cli-testsdk/azure/cli/testsdk/utilities.py#L132

## Solution

We need to release the latest `azure-devtools` and use it to correctly converts `bytes` to `str`:

https://github.com/Azure/azure-python-devtools/blob/master/src/azure_devtools/scenario_tests/preparers.py#L108

```py
body = str(request.body, 'utf-8') if isinstance(request.body, bytes) else str(request.body)
```

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/recordings/test_create_for_rbac_with_existing_kv_cert.yaml [hunk 1]
```diff
@@ -641,12 +450,12 @@ interactions:
...
+    body: '{"availableToOtherTenants": false, "homepage": "https://cli_test_sp_with_kv_existing_cert000001",
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/tests/latest/recordings/test_create_for_rbac_with_existing_kv_cert.yaml:453]: The weird `'b''b\'` has been removed by using the latest `azure-devtools`. 

```diff
- body: 'b''b\''{"availableToOtherTenants": false, "homepage": "https://cli_test_sp_with_kv_existing_cert000001",
+ body: '{"availableToOtherTenants": false, "homepage": "https://cli_test_sp_with_kv_existing_cert000001",
```

**Comment by yonzhan** [comment]
add to S173

**Comment by jiasli** [comment]
Merged into #14958.

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 2]
```diff
@@ -152,7 +152,7 @@ def default_api_version(self):
         ResourceType.MGMT_RESOURCE_DEPLOYMENTSCRIPTS: '2019-10-01-preview',
         ResourceType.MGMT_NETWORK_DNS: '2018-05-01',
         ResourceType.MGMT_KEYVAULT: '2019-09-01',
-        ResourceType.MGMT_AUTHORIZATION: SDKProfile('2018-09-01-preview', {
+        ResourceType.MGMT_AUTHORIZATION: SDKProfile('2020-04-01-preview', {
             'classic_administrators': '2015-06-01',
             'role_definitions': '2018-01-01-preview',
```
Comment (qwordy) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:157]: Do we need update this line?

**Review by qwordy — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py (hunk 2 above)
Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py:157]: `2018-01-01-preview` is the latest version which contains `RoleDefinition`.

File: src/azure-cli/azure/cli/command_modules/role/tests/hybrid_2018_03_01/test_role.py [hunk 3]
```diff
@@ -92,7 +93,7 @@ def test_create_for_rbac_with_cert_with_assignment(self, resource_group):
 
 class RbacSPKeyVaultScenarioTest2(ScenarioTest):
     @ResourceGroupPreparer(name_prefix='cli_test_sp_with_kv_new_cert')
-    @KeyVaultPreparer(name_prefix='test_create_for_rbac_with_new_kv_cert')
+    @KeyVaultPreparer(name_prefix='test-rbac-new-kv')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/tests/hybrid_2018_03_01/test_role.py:96]: The name is too long for `name_prefix`. Shortening it.

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py [hunk 4]
```diff
@@ -81,8 +81,31 @@ def test_native_app_create_scenario(self):
...
+
+        # TODO: https://github.com/Azure/azure-cli/pull/13769 fails to work
+        # Cert created with
+        # openssl req -newkey rsa:2048 -nodes -keyout key.pem -x509 -days 10000 -out certificate.pem
+        self.kwargs['cert_file'] = """
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:88]: Regenerate an un-expired cert.

File: src/azure-cli/azure/cli/command_modules/role/tests/util.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+from knack.util import CLIError
+
+
+def cmd_with_retry(self, *args, **kwargs):
+    """Retry self.cmd and checks until success."""
+    # Due to unstable role definition ARIs: https://github.com/Azure/azure-cli/issues/3187
+    import time
+    result = None
+    while not result:
```
[the remark this answers is not in the document]
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/role/tests/util.py:14]: I don't exactly know what number is suitable. The result is pretty unpredictable. It won't affect the CI as the tests will definitely succeeded. So I want to put it on hold for now.

**Review by zhoxing-ms — APPROVED** [review]

**Review by arrownj — APPROVED** [review]

**Review by MyronFanQiu — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/role/tests/util.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/role/tests/util.py:17]: Added as requested. Let's use `ROLE_COMMAND_MAX_RETRY = 20`  for now.

**Review by arrownj — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Pull request merged**